### ISSUE TITLE
实验复现

### ISSUE THREAD

**Issue by skdebris** [kwuking/TimeMixer#39]
按照给出的readme文件，执行scripts中weather的long term forecast 的 TimeMixer.sh 脚本，给出的训练和运行结果如图
![image](https://github.com/kwuking/TimeMixer/assets/163828054/01dfa3e5-f981-4ba6-afba-4c5247c8479e)
只是 scale = 0 的结果，应该修改哪个参数得到Mixed的准确预测结果

**Comment by kwuking** [comment]
你好，感谢对于我们工作的关注，目前最终结果的mixed我们使用的是sum的方式，您也可以尝试使用mean或者medium的方式进行计算，您只需要修改模型代码中最后关于dec_out的输出处理即可。

**Comment by yslbit** [comment]
> 按照给出的readme文件，执行scripts中weather的long term forecast 的 TimeMixer.sh 脚本，给出的训练和运行结果如图 ![image](https://private-user-images.githubusercontent.com/163828054/340890517-01dfa3e5-f981-4ba6-afba-4c5247c8479e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzODU0ODAsIm5iZiI6MTcxOTM4NTE4MCwicGF0aCI6Ii8xNjM4MjgwNTQvMzQwODkwNTE3LTAxZGZhM2U1LWY5ODEtNGJhNi1hZmJhLTRjNTI0N2M4NDc5ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQwNjU5NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZDE2M2IwMDEyODBkZTU1Nzk4YzU2MmM4YTJhODE3YzVhMGJmMWY5M2Q4OGMyMjE1MjMyYzI1ZGZiMjUzZjQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BcdlXQ0FKt1XJ17HCyTRHEoZaQV_RvG-wAZKIZIxqZA) 只是 scale = 0 的结果，应该修改哪个参数得到Mixed的准确预测结果

你好，你使用了他说最后混合方式方法嘛，我自己也验证了一个长时间序列预测的模型，结果和论文给出的结果相差很大

**Comment by kwuking** [comment]
> > 按照给出的readme文件，执行scripts中weather的long term forecast 的 TimeMixer.sh 脚本，给出的训练和运行结果如图 ![image](https://private-user-images.githubusercontent.com/163828054/340890517-01dfa3e5-f981-4ba6-afba-4c5247c8479e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzODU0ODAsIm5iZiI6MTcxOTM4NTE4MCwicGF0aCI6Ii8xNjM4MjgwNTQvMzQwODkwNTE3LTAxZGZhM2U1LWY5ODEtNGJhNi1hZmJhLTRjNTI0N2M4NDc5ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQwNjU5NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZDE2M2IwMDEyODBkZTU1Nzk4YzU2MmM4YTJhODE3YzVhMGJmMWY5M2Q4OGMyMjE1MjMyYzI1ZGZiMjUzZjQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BcdlXQ0FKt1XJ17HCyTRHEoZaQV_RvG-wAZKIZIxqZA) 只是 scale = 0 的结果，应该修改哪个参数得到Mixed的准确预测结果
> 
> 你好，你使用了他说最后混合方式方法嘛，我自己也验证了一个长时间序列预测的模型，结果和论文给出的结果相差很大

你好，请问你是在什么数据集验证的哪个一个长时间序列预测的模型呢？

**Comment by yslbit** [comment]
> > > 按照给出的readme文件，执行scripts中weather的long term forecast 的 TimeMixer.sh 脚本，给出的训练和运行结果如图 ![image](https://private-user-images.githubusercontent.com/163828054/340890517-01dfa3e5-f981-4ba6-afba-4c5247c8479e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzODU0ODAsIm5iZiI6MTcxOTM4NTE4MCwicGF0aCI6Ii8xNjM4MjgwNTQvMzQwODkwNTE3LTAxZGZhM2U1LWY5ODEtNGJhNi1hZmJhLTRjNTI0N2M4NDc5ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQwNjU5NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZDE2M2IwMDEyODBkZTU1Nzk4YzU2MmM4YTJhODE3YzVhMGJmMWY5M2Q4OGMyMjE1MjMyYzI1ZGZiMjUzZjQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BcdlXQ0FKt1XJ17HCyTRHEoZaQV_RvG-wAZKIZIxqZA) 只是 scale = 0 的结果，应该修改哪个参数得到Mixed的准确预测结果
> > 
> > 
> > 你好，你使用了他说最后混合方式方法嘛，我自己也验证了一个长时间序列预测的模型，结果和论文给出的结果相差很大
> 
> 你好，请问你是在什么数据集验证的哪个一个长时间序列预测的模型呢？

你好，我用的是long_term_forecast下面的ECL_script下面的脚本，得出来的指标和论文给出的相差很大

**Comment by skdebris** [comment]
> 你好，感谢对于我们工作的关注，目前最终结果的mixed我们使用的是sum的方式，您也可以尝试使用mean或者medium的方式进行计算，您只需要修改模型代码中最后关于dec_out的输出处理即可。

您好，感谢您的答复。我们找到了model代码中，关于dec_out的输出形式的代码，
dec_out = torch.stack(dec_out_list, dim=-1).sum(-1)
修改其中的sum为mean的方式进行计算，最终结果并没有太大的不同，test_result中画出的预测结果图仍然跟论文给出的相差较大。
[60.pdf](https://github.com/user-attachments/files/16078551/60.pdf)
图上的结果貌似并没有进行良好的多尺度预测的特征融合，想问下，该如何正确复现您论文中的结果。

**Comment by yslbit** [comment]
> > 你好，感谢对于我们工作的关注，目前最终结果的mixed我们使用的是sum的方式，您也可以尝试使用mean或者medium的方式进行计算，您只需要修改模型代码中最后关于dec_out的输出处理即可。
> 
> 您好，感谢您的答复。我们找到了model代码中，关于dec_out的输出形式的代码， dec_out = torch.stack(dec_out_list, dim=-1).sum(-1) 修改其中的sum为mean的方式进行计算，最终结果并没有太大的不同，test_result中画出的预测结果图仍然跟论文给出的相差较大。 [60.pdf](https://github.com/user-attachments/files/16078551/60.pdf) 图上的结果貌似并没有进行良好的多尺度预测的特征融合，想问下，该如何正确复现您论文中的结果。

我也想知道，大佬，求告知

**Comment by kwuking** [comment]
> 你好，我用的是long_term_forecast下面的ECL_script下面的脚本，得出来的指标和论文给出的相差很大

你好，请问是采用的相同的torch环境嘛，由于不同设备之间的差异，可能存在一定差别，但理论上是不应该有巨大差别的，相关细节你可以和我邮件联系。

**Comment by kwuking** [comment]
> weather

你好，请问是目前weather预测的指标效果不理想嘛？

**Comment by kwuking** [comment]
> > 你好，感谢对于我们工作的关注，目前最终结果的mixed我们使用的是sum的方式，您也可以尝试使用mean或者medium的方式进行计算，您只需要修改模型代码中最后关于dec_out的输出处理即可。
> 
> 您好，感谢您的答复。我们找到了model代码中，关于dec_out的输出形式的代码， dec_out = torch.stack(dec_out_list, dim=-1).sum(-1) 修改其中的sum为mean的方式进行计算，最终结果并没有太大的不同，test_result中画出的预测结果图仍然跟论文给出的相差较大。 [60.pdf](https://github.com/user-attachments/files/16078551/60.pdf) 图上的结果貌似并没有进行良好的多尺度预测的特征融合，想问下，该如何正确复现您论文中的结果。

你好，感谢对于我们工作的关注，确实weather数据集的数据规律非常杂乱，比较难以预测，想要得到非常理想的预测结果却是比较困难，我们目前是直接采用的sum方式进行融合，对于更好的融合方式我们也在进行探索，对于weather数据集而言，想要得到理想的融合结果对于训练的敏感度较高，您可以尝试动态调整学习率，训练更多的epoch，如果有进一步的问题，您可以和我邮件联系。
